aks.tf

```diff
@@ -0,0 +1,89 @@
+resource "azurerm_kubernetes_cluster" "aks" {
+  lifecycle {
+    ignore_changes = [
+      default_node_pool[0].node_count
+    ]
+  }
+
+  name                       = local.cluster_name
+  enable_pod_security_policy = false
+  location                   = local.resource_group.location
+  resource_group_name        = local.resource_group.name
+  dns_prefix                 = local.cluster_name
+  private_cluster_enabled    = false
+
+  api_server_authorized_ip_ranges = var.api_server_authorized_ip_ranges
+
+  default_node_pool {
+    enable_node_public_ip = var.node_pools.platform.enable_node_public_ip
+    name                  = "platform"
+    node_count            = var.node_pools.platform.min_count
+    node_labels           = var.node_pools.platform.node_labels
+    vm_size               = var.node_pools.platform.vm_size
+    availability_zones    = var.node_pools.platform.zones
+    os_disk_size_gb       = var.node_pools.platform.os_disk_size_gb
+    node_taints           = var.node_pools.platform.node_taints
+    enable_auto_scaling   = var.node_pools.platform.enable_auto_scaling
+    min_count             = var.node_pools.platform.min_count
+    max_count             = var.node_pools.platform.max_count
+    max_pods              = var.node_pools.platform.max_pods
+    tags                  = {}
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  addon_profile {
+    kube_dashboard {
+      enabled = false
+    }
+
+    oms_agent {
+      enabled                    = true
+      log_analytics_workspace_id = azurerm_log_analytics_workspace.logs.id
+    }
+  }
+
+  network_profile {
+    load_balancer_sku  = "Standard"
+    network_plugin     = "azure"
+    network_policy     = "calico"
+    dns_service_ip     = "100.97.0.10"
+    docker_bridge_cidr = "172.17.0.1/16"
+    service_cidr       = "100.97.0.0/16"
```

Comment:
how do we know these don't conflict with the cidr of the network?

Reply:
Do we even have control over the network cidr? Should these just be configurable anyway?

Reply:
Looks like we assume the network is pre-created. Why aren't we going with defaults or just looking at what is configured?

Reply:
Not pre-created, but I thought implicitly created by AKS in their own resource group. Any particular reason to support an existing vnet? 

Using their defaults makes sense, these values match our Rancher/AWS defaults which isn't a great reason.

cc @Secretions

Reply:
Going to split this out into a separate PR: https://github.com/dominodatalab/terraform-azure-aks/pull/14